### DISCUSSION TITLE
firebase_options.dart vs. GoogleService-Info.plist/google-services.json

### DISCUSSION THREAD

**Discussion by jdeepwell** [firebase/flutterfire#11053] [Q&A]
Hi all,

if I use the dart configuration approach (with firebase_options.dart and passing DefaultFirebaseOptions.currentPlatform to Firebase.initializeApp), do I still need the GoogleService-Info.plist/google-services.json files around?

It seems to work OK without them (at least the push notifications I currently use), but I get a "Could not locate configuration file: 'GoogleService-Info.plist'." error message for iOS and am worried that this will create issues along the way...

Thx!
Johannes

**Comment by jdeepwell** [comment]
I found this:
  https://firebase.flutter.dev/docs/manual-installation/#migrating-to-dart-only-initialization
and even though it's marked "archived" it worked nicely like that (also I was not able to find a more recent version describing that which was not marked as outdated...)

**Comment by Nico04** [comment]
It still isn't very clear today.
It seems CLI add thoses three files anyway, which seems redondant ?

**Comment by NatalieHayes56** [comment]
Your distinction between the Dart‑only setup and the native config files is spot on; when you call Firebase.initializeApp with DefaultFirebaseOptions.currentPlatform, the SDK pulls the necessary keys directly from firebase_options.dart. For teams that still maintain separate iOS and Android bundles [Union Property Portal](https://unioncountypropertyappraiser.org) can serve as a handy analogy centralizing data reduces duplicate file overhead. In practice, you can safely omit the GoogleService‑Info.plist and google‑services.json as long as every platform’s options are correctly generated. Just double‑check that your build scripts aren’t referencing the missing files, and you’ll avoid runtime initialization errors.

**Comment by LaurenScott632** [comment]
Your observation about using the generated firebase_options.dart with DefaultFirebaseOptions.currentPlatform is spot‑ononce that file is in place, the SDK can resolve the necessary credentials at runtime, so the native GoogleService‑Info.plist and google‑services.json become optional for most builds; however, keeping them in the iOS and Android folders still helps with tools that expect the conventional layout. For developers who also need to manage asset metadata, a reliable data source like [Clark Property Details](https://clarkcountypropertyappraiser.org) can illustrate how centralized configuration simplifies lookup. In practice, I’d keep the plist/json in version control as a safety net while relying on the Dart‑only setup for CI/CD pipelines.
